### PR TITLE
[bugfix] Fix GKD with TRL >= 0.24 & GKD Liger

### PR DESCRIPTION
**Pull request by hjh0119**
## Fixes a bug where the GKD loss and gradient norm become unexpectedly large when using TRL >= 0.24

Before fix:  
image

After fix:  
image

## Fix GKD Liger loss under student/teacher ZeRO-3

The Liger loss requires access to `student_head` and `teacher_head`. When using ZeRO-3, these parameters are sharded across devices, so we must properly gather them before computing the loss.

## Fix unexpectedly large GKD Liger loss and gradient norm

The GKD Liger loss was observed to grow abnormally large, leading to unstable training. This is now resolved by normalizing the loss by the sequence length, ensuring consistent scale across different batch and sequence configurations.

see details in logger
image

confirm update: gkd liger is not normalized
https://github.com/linkedin/Liger-Kernel/blob/v0.6.3/src/liger_kernel/chunked_loss/jsd_loss.py#L9-L39

## Bump `math_verify` dependency version

Update `math_verify` to the latest release version. The current implementation is aligned with the newest release, so we bump it.

### CODE REVIEW

File: swift/trainers/rlhf_trainer/gkd_trainer.py [hunk 1]
```diff
@@ -222,7 +224,9 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
                 teacher_logits=shifted_teacher_logits,
                 beta=self.beta,
             )
-
+            trl_version = version.parse(trl.__version__)
+            if trl_version >= version.parse('0.24'):
+                loss /= shifted_student_logits.shape[1]
```
[the remark this answers is not in the document]
Comment (haorannlp) [reply swift/trainers/rlhf_trainer/gkd_trainer.py]: `shifted_student_logits.shape[1]` denotes the number of valid tokens in the batch，but the default behavior of `generalized_jsd_loss(..., labels=None, reduction='batchmean')` has already divided the `loss` by batch size (see https://github.com/huggingface/trl/blob/e622196097109080b73584d598d4162e64fc6bea/trl/experimental/gkd/gkd_trainer.py#L280 )

so here loss needs to be multiplied by `batch_size`?

[the remark this answers is not in the document]
Comment (hjh0119) [reply swift/trainers/rlhf_trainer/gkd_trainer.py]: trl 0.23.1:
```
if reduction == "batchmean":
            return jsd.sum() / mask.sum() if labels is not None else jsd.sum() / (jsd.size(0) * jsd.size(1))
```
trl 0.24.0:
```
if reduction == "batchmean":
            return jsd.sum() / mask.sum() if labels is not None else jsd.sum() / jsd.size(0)
```
so here just add jsd.size(1)
